Lesson21/todo-app/sahra/script.js

```diff
@@ -0,0 +1,39 @@
+document
+  .getElementById("todoForm")
+  .addEventListener("submit", function (event) {
+    event.preventDefault();});
+
+  const todoInputElement = document.getElementById("todoInput");
+  const toDoListElement = document.getElementById("toDoList");
+  function addItem(){
+    if(todoInputElement.value === ''){
+      alert("There is nothing to do...");
+    } else {
+      const toDoListItemElement = document.createElement("li"); // Node
+      toDoListItemElement.textContent = todoInputElement.value;
+      toDoListElement.appendChild(toDoListItemElement);
+      let span = document.createElement("span");
+      span.innerHTML = "\u00d7";
+      toDoListItemElement.appendChild(span);
+    }
+    todoInputElement.value = "";
+    //to save data
+    saveData();
+  }
+
+  toDoListElement.addEventListener("click", function(e){
+    if(e.target.tagName === "LI"){
+      e.target.classList.toggle("completed");
+    } else if(e.target.tagName === "SPAN"){
+      e.target.parentElement.remove();
+    }
+  }, false);
+  //saving data function
+  function saveData(){
+    localStorage.setItem("data",toDoListElement.innerHTML);
+  }
+  /*to display everytime but mine doesn't completely works (checked buttons resets and removed items comes back)*/
+  function showTask(){
+    toDoListElement.innerHTML = localStorage.getItem("data");
```

Comment:
You only save task to the local storage when the task is added. You also need to update local storage if the task is marked/unmarked as completed or deleted.